golem-cli/src/clients/worker.rs

```diff
@@ -80,6 +83,14 @@ pub trait WorkerClient {
         name: WorkerName,
         template_id: RawTemplateId,
     ) -> Result<WorkerMetadata, GolemError>;
+    async fn find_metadata(
+        &self,
+        template_id: RawTemplateId,
+        filter: Option<Vec<String>>,
```

Comment:
I would rather call this as Option<workrfilter>. I don't think the call site need to be overly simplified to just pass vec<string>

Reply:
it is using this API https://github.com/golemcloud/golem/blob/main/openapi/golem-service.yaml#L63

and it is list of filters, in form `{property} {operator} {value}`, for example `name = worker-name`, we need to do some documentation ....